app/models/ingest_job.rb

```diff
@@ -994,7 +994,9 @@ def anndata_summary_props
     pipelines = ApplicationController.life_sciences_api_client.list_pipelines
     previous_jobs = pipelines.operations.select do |op|
       pipeline_args = op.metadata.dig('pipeline', 'actions').first['commands']
-      pipeline_args.detect { |c| c == study_file.id.to_s } && !pipeline_args.include?('--differential-expression')
+      op.done? &&
+        pipeline_args.detect { |c| c == study_file.id.to_s } &&
+        (!pipeline_args.include?('--differential-expression') && !pipeline_args.include?('--subsample'))
```

Comment:
Do we have any central / const-like variable that records "special actions" like jobs for DE or subsampling?  If we anticipate new such pipeline arguments for CAS or on-demand pairwise DE, then I could see not having that being a likely cause of issues down the line.

That's far from a blocking issue, just at least a note to keep in the back of our minds.

Reply:
Sort of?  We have the below constant for "special actions", but it's used to determine which jobs don't email users:

```
SPECIAL_ACTIONS = %i[differential_expression render_expression_arrays image_pipeline].freeze
```

I think your comment still stands though, and I can rewrite this as an inclusive check for the main three ingest processes (expression, metadata, clustering).

Reply:
[72544ff](https://github.com/broadinstitute/single_cell_portal_core/pull/2144/commits/72544ffdb774ef6f623567c15b2acd79be7eff70)